streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -564,8 +576,15 @@ private void restoreChangelog(final ChangelogMetadata changelogMetadata) {
             return Collections.emptyMap();
 
         try {
-            return restoreConsumer.endOffsets(partitions);
-        } catch (final TimeoutException e) {
+            if (adminClient != null) {
```

Comment:
Why do we need this distinction? Seems we set `adminClient` in any case and it should never be `null`?

Reply:
I also do not understand the distinction. When would we want to call `restoreConsumer.endOffsets(partitions)`?

Reply:
req: Could you add a test (or adapt an existing one) and verify whether during `restore()` a call to `adminClient.listOffsets()` with isolation level `READ_UNCOMMITTED` is done?

Reply:
This is just a hack-around: I will always use admin-client by passing it via the constructor.

I will add a unit test.

Reply:
Should we remove this check then?

Reply:
I think we can remove this `null` check? If you really want to add one, add it to the constructor?
```
this.adminClient = Objects.notNull(adminClient);
```

Reply:
I just saw that we kinda need to pass `null` as `AdminClient` in TDD -- however, `StoreChangelogReader#restore()` is never called by TTD, and thus I still think we should remove the `null` check and fail hard with a NPE as it would indicate a bug if `adminClient` is `null` in a regular deployment.